pinkopy/base_session.py

```diff
@@ -2,52 +2,53 @@
 import inspect
 import logging
 import time
+
 try:
     from urllib.parse import urlencode, urljoin
 except ImportError:
     from urllib import urlencode
     from urlparse import urljoin
-import xmltodict
 
+import xmltodict
 from cachetools.func import ttl_cache
 import requests
-
-from .exceptions import PinkopyError, raise_requests_error
+from pinkopy.exceptions import PinkopyError, raise_requests_error
 
 log = logging.getLogger(__name__)
 
+class BaseSession:
+    """
+    BaseSession class for interacting with the Commvault API.
 
-class BaseSession(object):
-    """BaseSession
-
-    This will not be instantiated directly. Other classes will inherit from this.
+    This class will not be instantiated directly. Other classes will inherit from this.
 
     Args:
-        service (optional[str]): URL and path to root of api
-        user (str): Commvault username
-        pw (str): Commvault password
-        use_cache (optional[bool]): Use cache? Defaults to False
-        cache_ttl (optional[int]): Duration cache lives. Defaults to 1200.
-        cache_methods (optional[int]): List of methods to cache.
-            Defaults provided by the inheriting classes.
-        token (optional[str]): Authtoken for header
+        service (str): URL and path to root of API.
+        user (str): Commvault username.
+        pw (str): Commvault password.
+        use_cache (bool, optional): Use cache? Defaults to True.
+        cache_ttl (int, optional): Duration cache lives. Defaults to 1200 seconds.
+        cache_methods (list, optional): List of methods to cache. Defaults to an empty list.
+        token (str, optional): Auth token for headers.
 
     Returns:
-        session object
+        BaseSession: An instance of BaseSession or its subclass.
     """
-    def __init__(self, service, user, pw, use_cache=True, cache_ttl=1200,
-                 cache_methods=None, token=None):
+
+    def __init__(self, service, user, pw, use_cache=True, cache_ttl=1200, cache_methods=None, token=None):
         self.service = service
         self.user = user
         self.pw = pw
         self.headers = {
             'Authtoken': token,
             'Accept': 'application/json',
-            'Content-type': 'application/json'
+            'Content-Type': 'application/json'
         }
+
         if not self.headers['Authtoken']:
             self.get_token()
-        self.__use_cache = bool(use_cache)
```

Comment:
The reason this used bool, was to handle truthy and non-truthy values like `"false"`. Is that problematic?